Wrap GCVSplineSet

klshrinidhi opened the pull request:
Addressing issue #1622.

chrisdembia commented:
It would be good to test the specific constructor that we want to have working in the bindings. Also, it would be great, as @aseth1 mentioned, to add a constructor that takes a TimeSeriesTable. For my research, I've written such a function:


```cpp
GCVSplineSet createGCVSplineSet(const TimeSeriesTable& table,
                                const std::vector<std::string>& labels = {},
                                int degree = 5,
                                double errorVariance = 0.0) {
    GCVSplineSet set;
    const auto& time = table.getIndependentColumn();
    auto labelsToUse = labels;
    if (labelsToUse.empty()) labelsToUse = table.getColumnLabels();
    for (const auto& label : labelsToUse) {
        const auto& column = table.getDependentColumn(label);
        set.adoptAndAppend(new GCVSpline(degree, column.size(), time.data(),
                                         &column[0], label, errorVariance));
    }
    return set;
}
```

klshrinidhi commented:
* I have added constructor to take a TimeSeriesTable as per your code snippet.
* I have added a small test to make sure the constructors are callable.

chrisdembia commented:
Looks great. Only one comment: could you move the new doxygen block to the header file?

klshrinidhi commented:
I have moved all the doxygen comments to header file to be consistent.

chrisdembia commented on OpenSim/Common/GCVSplineSet.h:
```diff
@@ -49,46 +49,113 @@ class Storage;
+     *
+     * @param aIndex Index of the desired function:  0 <= aIndex < getSize().
+     * @return Function at index aIndex.  If aIndex is not value NULL is 
+     *         returned.
```
Maybe edit this to "If aIndex is not a valid value..."

klshrinidhi replied:
Done.
